### ISSUE TITLE
Type-Declarations with object-exports and namespaced classes can't be written

### ISSUE THREAD

**Issue by manuth** [microsoft/TypeScript#32090]
**TypeScript Version:**  3.4.0-dev.201xxxxx
**Search Terms:**
  * Export in namespaces
  * namespace merging
  * type-declarations
  * es6-namespaces
  * es6-modules

**General**
Some time ago I tried to write type-declarations for a module called `inquirer`.
This module is semi-namespace-structured.
While the exported object (`export = inquirer;`) is an object of type `PromptModule` it also provides some interfaces and classes in nested sub-properties.

Sadly TypeScript doesn't seem to allow one to write declarations for such a module.

**Code**
```ts
declare namespace inquirer {
    interface Inquirer extends PromptModule { }
    namespace ui {
        class BaseUI { /* ... */ }
    }
}

declare var inquirer: inquirer.Inquirer;
export = inquirer;
```

**Expected behavior:**
As soon as I change the keyword from `interface BaseUI` to `class BaseUI` TypeScript reports a `duplicate identifier`-error.

But the exported object `inquirer` _is_ an object of type `PromptModule` and `inquirer.ui.BaseUI` _is_ a class to instanciate. I can't write a declaration for it though.

**Actual behavior:**
TypeScript should not complain and just merge the variable and the namespace instead.

**Related Issues:**
https://stackoverflow.com/questions/56702764/how-to-write-type-declarations-for-namespace-structured-modules

**Comment by manuth** [comment]
https://github.com/microsoft/TypeScript/issues/13536
